lib/client.c

```diff
@@ -1099,11 +1190,35 @@ SF_STATUS snowflake_bind_param_array(
     }
     clear_snowflake_error(&sfstmt->error);
     if (sfstmt->params == NULL) {
-        sfstmt->params = sf_array_list_init();
+        sf_param_store_init(_sf_get_param_style(&sfbind_array[0]),
+                            &sfstmt->params);
     }
     for (i = 0; i < size; i++) {
-        sf_array_list_set(sfstmt->params, &sfbind_array[i],
-                          sfbind_array[i].idx);
+        sf_param_store_set(sfstmt->params, &sfbind_array[i],
+                sfbind_array[i].idx, sfbind_array[i].name);
+        /*
+         * For named parameters, populate the name list in the
+         * sfstmt to help extract the params from the treemap
+         * during execute
+         */
+        if (_sf_get_current_param_style(sfstmt) == NAMED)
+        {
+            sfstmt->name_list = SF_REALLOC(sfstmt->name_list,
```

Comment:
You could move this outside of the for loop and just do 1 memory allocation instead of many allocations.